shared/src/commonMain/kotlin/com/devil/phoenixproject/data/repository/UserProfileRepository.kt

```
@@ -153,9 +153,19 @@ class SqlDelightUserProfileRepository(private val database: VitruvianDatabase) :
-        gamificationRepository.saveRpgProfile(RpgAttributeEngine.computeProfile(rpgInput), targetProfileId)
+        // Issue #393: Wrap post-deletion gamification recompute in try-catch.
+        // getRpgInput() has no internal error handling; if it throws (e.g., missing
+        // data after cascade reassignment), the whole deleteProfile call propagates
+        // the exception to the UI, which on iOS causes SIGABRT.
+        try {
+            gamificationRepository.updateStats(targetProfileId)
+            val rpgInput = gamificationRepository.getRpgInput(targetProfileId)
+            gamificationRepository.saveRpgProfile(RpgAttributeEngine.computeProfile(rpgInput), targetProfileId)
+        } catch (e: kotlin.coroutines.cancellation.CancellationException) {
+            throw e
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)